packages/context/src/context.ts

```diff
@@ -259,10 +271,32 @@ export class Context {
     }
 
     if (isPromise(boundValue)) {
-      return boundValue.then(v => getDeepProperty(v, path));
+      return boundValue.then(v => Binding.getDeepProperty(v, path));
     }
 
-    return getDeepProperty(boundValue, path);
+    return Binding.getDeepProperty(boundValue, path);
+  }
+
+  /**
+   * Clone the context with an optional new parent context
+   * @param parent Optional parent context
+   */
+  clone(parent?: Context) {
+    const copy = new Context();
+    copy._parent = parent || this._parent;
+    copy.registry = new Map(this.registry);
+    return copy;
+  }
+
+  /**
+   * Merge bindings from the given context into this one
+   * @param ctx Another context
+   */
+  mergeWith(ctx: Context) {
```

Comment:
I find the name `mergeWith` misleading, because the method is merging only the bindings registered in this particular context, but ignoring any entries inherited from parent contexts.

An example to illustrate my point:

```ts
const app = new Context();
app.bind('http.port').to(8080);

const component = new Context(app);
class MyClient {
  constructor(@inject('http.port') private port: number) {}
}
app.bind('http.client').toClass(MyClient);

const ctx1 = new Context();
ctx1.mergeWith(component);
await ctx.get('http.client');
// error, "http.port" was not bound
```